server/src/main/java/com/genymobile/scrcpy/ScreenCapture.java

```diff
@@ -100,8 +101,11 @@ public void onRotationChanged(int rotation) {
     private static IBinder createDisplay() throws Exception {
         // Since Android 12 (preview), secure displays could not be created with shell permissions anymore.
         // On Android 12 preview, SDK_INT is still R (not S), but CODENAME is "S".
-        boolean secure = Build.VERSION.SDK_INT < Build.VERSION_CODES.R || (Build.VERSION.SDK_INT == Build.VERSION_CODES.R && !"S".equals(
-                Build.VERSION.CODENAME));
+        boolean secure = Build.VERSION.SDK_INT < Build.VERSION_CODES.R || (Build.VERSION.SDK_INT == Build.VERSION_CODES.R && !"S"
+                .equals(Build.VERSION.CODENAME));
+        if (Os.getuid() == 1000 || Os.getuid() == 1003) {
```

Comment:
Perhaps add this to the condition above instead of mutating the variable (and edit the comment accordingly)?
Just a suggestion, not sure if it gets too confusing.

Reply:
I simply patched in the original check without mangling it, but now I did refactor it to include the new EUID check as part of the initial selection procedure.